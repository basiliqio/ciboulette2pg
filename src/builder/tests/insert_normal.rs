use super::*;

#[test]
fn simple() {
    let mut builder = Ciboulette2PostgresBuilder::new();
    let table = CibouletteTableSettings::new(
        "id".to_string(),
        "uuid".to_string(),
        "public".to_string(),
        "mysimpletable".to_string(),
    );
    let params: Vec<(&str, Ciboulette2SqlValue<'_>)> = vec![
        (
            "first-name",
            Ciboulette2SqlValue::Text(Some(Cow::Borrowed("hello"))),
        ),
        (
            "last-name",
            Ciboulette2SqlValue::Text(Some(Cow::Borrowed("world"))),
        ),
    ];
    builder.gen_insert_normal(&table, params, true).unwrap();
    let res = builder.build().unwrap();

    insta::assert_debug_snapshot!(res);
}

#[test]
fn no_returning() {
    let mut builder = Ciboulette2PostgresBuilder::new();
    let table = CibouletteTableSettings::new(
        "id".to_string(),
        "uuid".to_string(),
        "public".to_string(),
        "mysimpletable".to_string(),
    );
    let params: Vec<(&str, Ciboulette2SqlValue<'_>)> = vec![
        (
            "first-name",
            Ciboulette2SqlValue::Text(Some(Cow::Borrowed("hello"))),
        ),
        (
            "last-name",
            Ciboulette2SqlValue::Text(Some(Cow::Borrowed("world"))),
        ),
    ];
    builder.gen_insert_normal(&table, params, false).unwrap();
    let res = builder.build().unwrap();

    insta::assert_debug_snapshot!(res);
}

#[test]
fn no_params() {
    let mut builder = Ciboulette2PostgresBuilder::new();
    let table = CibouletteTableSettings::new(
        "id".to_string(),
        "uuid".to_string(),
        "public".to_string(),
        "mysimpletable".to_string(),
    );
    builder.gen_insert_normal(&table, vec![], true).unwrap();
    let res = builder.build().unwrap();

    insta::assert_debug_snapshot!(res);
}
