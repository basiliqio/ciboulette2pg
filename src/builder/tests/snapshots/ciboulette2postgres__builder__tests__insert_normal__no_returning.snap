---
source: src/builder/tests/insert_normal.rs
expression: res

---
(
    "INSERT INTO \"public\".\"mysimpletable\" (\"first-name\", \"last-name\") VALUES ($0, $1)",
    Ciboulette2SqlArguments {
        inner: [
            Text(
                Some(
                    "hello",
                ),
            ),
            Text(
                Some(
                    "world",
                ),
            ),
        ],
    },
)
